lib/miq_automation_engine/engine/miq_ae_engine.rb

```diff
@@ -318,6 +319,7 @@ def self.resolve_automation_object(uri, user_obj, attr = nil, options = {}, read
     raise "User object not passed in" unless user_obj.kind_of?(User)
     uri = create_automation_object(uri, attr, options) if attr
     options[:uri] = uri
+    User.current_user = user_obj
```

Comment:
@mkanoor this is a global variable (thread local)
Is there a chance that automate will call into here from itself and modify this value?